### PR TITLE
Add original Brzozowski paper and start porting it to Coq with some proofs.

### PR DESCRIPTION
**Pull request by awalterschulze**
The idea is to learn things in this folder, that we can bring back to the top folder.
The idea is not to import things from this folder into the top folder or vice versa.

The .v file is trying to translate the paper into Coq, in the most accurate way possible.

There are many problems and this is probably not ready to be merged.

### CODE REVIEW

**Comment by awalterschulze** [comment]
Maybe this would be a better description of a derivative, but it seems so complicated

```coq
Inductive derivative: regex -> string -> regex -> Prop :=
    | is_derivative (r: regex) (s: string) (dr: regex):
        (exists (dr: regex),
            forall (t: string)
                   (m: is_member r (s ++ t)),
            is_member dr t
        ) -> derivative r s dr
    .
```

This is the original definition from the paper
> We define another operation on a set $R$ of sequences, 
    yielding a new set of sequences called a derivative of $R$.
    **Definition 3.1.** Given a set $R$ of sequences and a finite sequence $s$, 
    the derivative of $R$ with respect to $s$ is denoted by $D_s R$ and is 
    $D_s R = \{t | s.t \in R \}$.

**Comment by paulcadman** [comment]
This definition of derivative is from the point of view that a regular expression *is* the set of strings that it matches.

In this view - the derivative of a regular expression `r` at `s` is the set consisting of strings `t` such that `s ++ t` is in the set of strings that `r` matches.

**Comment by awalterschulze** [comment]
That is a great way to put it.
Do you think the above Inductive predicate formalises what you have said?

**Comment by awalterschulze** [comment]
The is_member and not_member seem to be working.
I decided to punt the derivative definition for a following pull request.

File: Makefile.conf [hunk 1]
```diff
@@ -35,9 +35,9 @@ COQMF_CMDLINE_COQLIBS =
 ###############################################################################
 
 COQMF_LOCAL=0
-COQMF_COQLIB=/usr/local/lib/coq/
-COQMF_DOCDIR=/usr/local/Cellar/coq/8.11.1/share/doc/coq/
-COQMF_OCAMLFIND=/usr/local/opt/ocaml-findlib/bin/ocamlfind
+COQMF_COQLIB=/Applications/CoqIDE_8.11.1.app/Contents/Resources/lib/coq/
```
Comment (aymanosman) [inline Makefile.conf:38]: This file obviously needs to be ignored

Comment (awalterschulze) [reply Makefile.conf:38]: Lets make a follow pull request for that.

**Pull request merged**